crates/ruff_linter/src/rules/pyupgrade/rules/pep695/class_with_mixed_type_vars.rs

```diff
@@ -0,0 +1,204 @@
+use std::iter;
+
+use crate::checkers::ast::Checker;
+use crate::fix::edits::{remove_argument, Parentheses};
+use crate::rules::pyupgrade::rules::pep695::{
+    expr_name_to_type_var, find_generic, DisplayTypeVars, TypeParamKind, TypeVar,
+};
+use crate::settings::types::PythonVersion;
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Arguments, Expr, ExprSubscript, ExprTuple, StmtClassDef, TypeParams};
+use ruff_python_semantic::{Binding, BindingKind, SemanticModel};
+
+/// ## What it does
+/// Checks for classes that both have PEP 695 type parameter list
+/// and inherit from `typing.Generic` or `typing_extensions.Generic`.
+///
+/// ## Why is this bad?
+/// Such classes cause errors at runtime:
+///
+/// ```python
+/// from typing import Generic, TypeVar
+///
+/// U = TypeVar("U")
+///
+/// # TypeError: Cannot inherit from Generic[...] multiple times.
+/// class C[T](Generic[U]): ...
+/// ```
+///
+/// ## Example
+///
+/// ```python
+/// from typing import Generic, ParamSpec, TypeVar, TypeVarTuple
+///
+/// U = TypeVar("U")
+/// P = ParamSpec("P")
+/// Ts = TypeVarTuple("Ts")
+///
+///
+/// class C[T](Generic[U, P, Ts]): ...
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// class C[T, U, **P, *Ts]: ...
+/// ```
+///
+/// ## Fix safety
+/// As the fix changes runtime behaviour, it is always marked as unsafe.
+#[derive(ViolationMetadata)]
+pub(crate) struct ClassWithMixedTypeVars;
+
+impl Violation for ClassWithMixedTypeVars {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Class with type parameter list inherits from `Generic`".to_string()
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Convert to new-style".to_string())
```

Comment:
This doesn't feel very descriptive about what the autofix actually does... maybe this?

```suggestion
        Some("Remove `Generic` base class".to_string())
```